### PR TITLE
Initial version of graph simplification.

### PR DESCRIPTION
**Pull request by bkidney**
Implements initial version of graph simplification based
 loosely on the SOAAP work.

### CODE REVIEW

File: cgnet/simplify.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import networkx as nx
```
Comment (trombonehero) [inline cgnet/simplify.py:23]: Copyright header?

File: cgnet/simplify.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+def is_simple_node( graph, node ):
```
Comment (trombonehero) [inline cgnet/simplify.py]: I don't think that there should be spaces around parameters (see [PEP 8](https://www.python.org/dev/peps/pep-0008/#pet-peeves)).

File: cgnet/simplify.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    """A node is "Simple" if none of the following is true
+    - it has multiple inputs (it joins chains together)
+    - it has no inputs (it's a root node)
+    - it has multiple outputs (it splits chains apart)
+    - it has no outputs (it's a leaf node)
+
+    Keyword arguments:
+    node -- A networkx DiGraph Node
+    """
+    return graph.in_degree(node) == 1 \
```
Comment (trombonehero) [inline cgnet/simplify.py]: Might this fit in 80 cols?

File: cgnet/simplify.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            and graph.out_degree(node)  == 1
```
Comment (trombonehero) [inline cgnet/simplify.py]: Extra space before `==`

File: cgnet/simplify.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+def simplify_graph( graph ):
```
Comment (trombonehero) [inline cgnet/simplify.py]: I think I'd name this something like `simplified` to emphasize that we aren't modifying the original graph.

File: cgnet/simplify.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    """Simplify a CallGraph by collapsing call chains and dropping
```
Comment (trombonehero) [inline cgnet/simplify.py]: Maybe "coalescing call chains"? Perhaps this language is taken from the go-soaap work, in which case, I'm sorry I didn't think of *le mot juste* sooner! :)

File: cgnet/utils.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (trombonehero) [inline cgnet/utils.py:22]: Copyright, spacing, etc., as mentioned in `simplify.py`.

File: cgnet/simplify.py (hunk 1 above)
Comment (bkidney) [reply cgnet/simplify.py:23]: Which one should I use? CADETS?

Comment (trombonehero) [reply cgnet/simplify.py:23]: Yup, that would probably be reasonable.

Comment (bkidney) [reply cgnet/simplify.py:23]: Fixed.

File: cgnet/utils.py (hunk 7 above)
Comment (bkidney) [reply cgnet/utils.py:22]: Fixed.

**Pull request merged**